### PR TITLE
Unittests

### PR DESCRIPTION
**Pull request by gertjanvanzwieten**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -33,35 +33,30 @@ jobs:
...
+          - {name: "python 3.5", os: ubuntu-latest, python-version: "3.5", matrix-backend: numpy, nprocs: 1}
+          - {name: "python 3.6", os: ubuntu-latest, python-version: "3.6", matrix-backend: numpy, nprocs: 1}
+          - {name: "python 3.7", os: ubuntu-latest, python-version: "3.7", matrix-backend: numpy, nprocs: 1}
+          - {name: "python 3.8", os: ubuntu-latest, python-version: "3.8", matrix-backend: numpy, nprocs: 1}
+          - {name: "python 3.9", os: ubuntu-latest, python-version: "3.9", matrix-backend: numpy, nprocs: 1}
+          - {name: "scipy matrix", os: ubuntu-latest, python-version: "3.10", matrix-backend: scipy, nprocs: 1}
+          - {name: "mkl matrix", os: ubuntu-latest, python-version: "3.10", matrix-backend: mkl, nprocs: 1}
+          - {name: "mkl matrix parallel", os: ubuntu-latest, python-version: "3.10", matrix-backend: mkl, nprocs: 2}
+          - {name: "parallel", os: ubuntu-latest, python-version: "3.10", matrix-backend: numpy, nprocs: 2}
+          - {name: "numpy 1.17", os: ubuntu-latest, python-version: 3.6, matrix-backend: numpy, nprocs: 1, numpy-version: ==1.17}
```
Comment (joostvanzwieten) [inline .github/workflows/test.yaml:53]: Suggestion: surround `3.6` with quotes.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -86,42 +83,26 @@ jobs:
...
+          python -um devtools.gha.configure_mkl
       - name: Test
-        env:
-          NUTILS_DEBUG: all
         run: |
-          mkdir testenv
-          cp -r examples docs tests .coveragerc testenv
-          cd testenv
-          python -um coverage run -m unittest -bq
+          python -um coverage run -m unittest -bq tests/test_*.py
```
Comment (joostvanzwieten) [inline .github/workflows/test.yaml:103]: Is listing `tests/test_*.py` necessary if we remove `examples/__init__.py`?

**Review by joostvanzwieten — APPROVED** [review]

File: .github/workflows/test.yaml (hunk 2 above)
Comment (gertjanvanzwieten) [reply .github/workflows/test.yaml:103]: Without this, python <= 3.6 stumbles over import dataclasses when it enters devtools during discovery...

Comment (gertjanvanzwieten) [reply .github/workflows/test.yaml:103]: I don't like this much either

Comment (joostvanzwieten) [reply .github/workflows/test.yaml:103]: What do you think of `python -m unittest discover -t . -s test`? Then you can also omit removing `examples/__init__.py`.

Comment (gertjanvanzwieten) [reply .github/workflows/test.yaml]: done

File: .github/workflows/test.yaml (hunk 1 above)
Comment (gertjanvanzwieten) [reply .github/workflows/test.yaml]: done

File: nutils/mesh.py [hunk 3]
```diff
@@ -361,7 +361,8 @@ def parsegmsh(mshdata):
       if keep.all():
         renum = numpy.arange(len(cells.data))
       else:
-        msh.cells[icell] = type(cells)(type=cells.type, data=cells.data[numpy.hstack([True, keep])])
+        print(type(cells))
```
Comment (joostvanzwieten) [inline nutils/mesh.py:364]: This looks like a debug statement.

**Review by joostvanzwieten — CHANGES_REQUESTED** [review]

File: nutils/mesh.py (hunk 3 above)
Comment (joostvanzwieten) [reply nutils/mesh.py]: fixed

**Review by joostvanzwieten — APPROVED** [review]

**Pull request merged**